crates/ruff_server/src/server.rs

```diff
@@ -132,22 +130,21 @@ impl Server {
         worker_threads: NonZeroUsize,
     ) -> crate::Result<()> {
         let mut scheduler =
-            schedule::Scheduler::new(&mut session, worker_threads, &connection.sender);
+            schedule::Scheduler::new(&mut session, worker_threads, connection.make_sender());
```

Comment:
Is it necessary to use a `WeakRef` here? I would expect that we can use a strong reference here because 
a) `connection` always outlives this function call`
b) `scheduler` gets dropped at the end of the function call (which exits when it is a shutdown message)

Reply:
We technically could use a strong reference but I'd rather enforce a weak-reference-only policy to guarantee that we don't have other strong references on `Connection::close`.